### PR TITLE
5.x merge 4.x

### PR DESCRIPTION
**Pull request by asmorkalov**
No changes in opencv_contrib and opencv_extra.

Manual changes:
- Fixed f16 in universal intrinsics for ARM, e.g. `template <> inline v_float16x8 v_setall_(float v) { return v_setall_f16((hfloat)v); }` overload.
- Data type fixes in fp16 test for v_exp, v_sin, v_cos, v_sincos, etc.

#25891 from WanliZhong:v_func-def-bug
#25892 from WanliZhong:v_sincos
#25909 from gblikas:patch-1
#26163 from asmorkalov:as/HAL_schaar_deriv
#26166 from mshabunin:fix-intrin-ops
#26176 from najasnake12:fixed_minor_typos_in_js_tutorials
#26187 from inayd:26130-fixFillPolyBoundaries
#26211 from Kumataro:fix26207
#26212 from jamacias:feature/TickMeter-lasttime
#26216 from hanliutong:rvv-hal-merge
#26219 from mlourakis:4.x
#26224 from mshabunin:cpp-videoio-backport
#26228 from mshabunin:cpp-features2d-4x
#26230 from mshabunin:cpp-photo-4x
#26234 from zachlowry:apply-gcc6-fix-on-each-directory
#26236 from asmorkalov:as/HAL_pyrlk_hack_documentation
#26241 from asmorkalov:as/kelidicv-0.2
#26245 from cudawarped:cuda_update_to_npp_stream_ctx
#26259 from Kumataro:fix26258
#26260 from sturkmen72:upd_doc_4_x
#26263 from mlourakis:4.x
#26266 from mshabunin:fix-rvv071-build
#26278 from Quantizs:feature-create-face-recognizer-from-buffer
#26281 from kallaballa:clgl_device_discovery
#26284 from fzuuzf:enum_arithmetic_fixes_for_c++26
#26293 from SeptimiuIoachimNeagaIntel:EISW-140103_optimization_flag
#26298 from sturkmen72:avif
#26313 from FantasqueX:ipp-warp-affine-border-value
#26320 from mshabunin:fix-cmake-in-list
#26335 from migueldaipre:4.x
#26340 from Kumataro:wa26339
#26353 from asmorkalov:as/ade_1.2e

Previous "Merge 4.x": #26135

### CODE REVIEW

**Comment by asmorkalov** [comment]
@vpisarev @WanliZhong Could you take a look on FP16 related changes in universal intrinsics. I tuned test types a bit and added overload `template <> inline v_float16x8 v_setall_(float v) { return v_setall_f16((hfloat)v); }`.

**Comment by WanliZhong** [comment]
> @vpisarev @WanliZhong Could you take a look on FP16 related changes in universal intrinsics. I tuned test types a bit and added overload `template <> inline v_float16x8 v_setall_(float v) { return v_setall_f16((hfloat)v); }`.

I think the implementation code need to be changed, I'm fixing it now.

File: modules/core/include/opencv2/core/hal/intrin_math.hpp [hunk 1]
```diff
@@ -34,434 +30,658 @@ namespace CV__SIMD_NAMESPACE {
...
+    const _TpVec16F _vexp_one_fp16 = v_setall_<_TpVec16F>(1.f);
+    const _TpVec16F _vexp_LOG2EF_f16 = v_setall_<_TpVec16F>(1.44269504088896341f);
+    const _TpVec16F _vexp_C1_f16 = v_setall_<_TpVec16F>(-6.93359375E-1f);
+    const _TpVec16F _vexp_C2_f16 = v_setall_<_TpVec16F>(2.12194440E-4f);
+    const _TpVec16F _vexp_p0_f16 = v_setall_<_TpVec16F>(1.9875691500E-4f);
+    const _TpVec16F _vexp_p1_f16 = v_setall_<_TpVec16F>(1.3981999507E-3f);
+    const _TpVec16F _vexp_p2_f16 = v_setall_<_TpVec16F>(8.3334519073E-3f);
+    const _TpVec16F _vexp_p3_f16 = v_setall_<_TpVec16F>(4.1665795894E-2f);
+    const _TpVec16F _vexp_p4_f16 = v_setall_<_TpVec16F>(1.6666665459E-1f);
+    const _TpVec16F _vexp_p5_f16 = v_setall_<_TpVec16F>(5.0000001201E-1f);
```
Comment (WanliZhong) [inline modules/core/include/opencv2/core/hal/intrin_math.hpp:53]: use `hfloat(xxxx.f)` to replace the input number
`v_setall_<_TpVec16F>(hfloat(-6.93359375E-1f));`

File: modules/core/include/opencv2/core/hal/intrin_math.hpp [hunk 2]
```diff
@@ -34,434 +30,658 @@ namespace CV__SIMD_NAMESPACE {
...
+    const _TpVec16F _vlog_q2_fp16 = v_setall_<_TpVec16F>(0.693359375f);
+    const _TpVec16F _vlog_p0_fp16 = v_setall_<_TpVec16F>(7.0376836292E-2f);
+    const _TpVec16F _vlog_p1_fp16 = v_setall_<_TpVec16F>(-1.1514610310E-1f);
+    const _TpVec16F _vlog_p2_fp16 = v_setall_<_TpVec16F>(1.1676998740E-1f);
+    const _TpVec16F _vlog_p3_fp16 = v_setall_<_TpVec16F>(-1.2420140846E-1f);
+    const _TpVec16F _vlog_p4_fp16 = v_setall_<_TpVec16F>(1.4249322787E-1f);
+    const _TpVec16F _vlog_p5_fp16 = v_setall_<_TpVec16F>(-1.6668057665E-1f);
+    const _TpVec16F _vlog_p6_fp16 = v_setall_<_TpVec16F>(2.0000714765E-1f);
+    const _TpVec16F _vlog_p7_fp16 = v_setall_<_TpVec16F>(-2.4999993993E-1f);
+    const _TpVec16F _vlog_p8_fp16 = v_setall_<_TpVec16F>(3.3333331174E-1f);
```
Comment (WanliZhong) [inline modules/core/include/opencv2/core/hal/intrin_math.hpp:207]: The same as above

File: modules/core/include/opencv2/core/hal/intrin_math.hpp [hunk 3]
```diff
@@ -34,434 +30,658 @@ namespace CV__SIMD_NAMESPACE {
...
+    _vlog_y = v_fma(_vlog_y, _vlog_x, _vlog_p5_fp16);
+    _vlog_y = v_fma(_vlog_y, _vlog_x, _vlog_p6_fp16);
+    _vlog_y = v_fma(_vlog_y, _vlog_x, _vlog_p7_fp16);
+    _vlog_y = v_fma(_vlog_y, _vlog_x, _vlog_p8_fp16);
+    _vlog_y = v_mul(_vlog_y, _vlog_x);
+    _vlog_y = v_mul(_vlog_y, _vlog_z);
+
+    _vlog_y = v_fma(_vlog_e, _vlog_q1_fp16, _vlog_y);
+
+    _vlog_y = v_sub(_vlog_y, v_mul(_vlog_z, v_setall_<_TpVec16F>(0.5f)));
```
Comment (WanliZhong) [inline modules/core/include/opencv2/core/hal/intrin_math.hpp:246]: The same as above

File: modules/core/include/opencv2/core/hal/intrin_math.hpp [hunk 4]
```diff
@@ -34,434 +30,658 @@ namespace CV__SIMD_NAMESPACE {
...
+
+    _TpVec16F _vlog_x, _vlog_e, _vlog_y, _vlog_z, _vlog_tmp;
+    _TpVec16S _vlog_ux, _vlog_emm0;
+    const _TpVec16S _vlog_inv_mant_mask_s16 = v_setall_<_TpVec16S>((short)~0x7c00);
+
+    _vlog_ux = v_reinterpret_as_s16(x);
+    _vlog_emm0 = v_shr(_vlog_ux, 10);
+
+    _vlog_ux = v_and(_vlog_ux, _vlog_inv_mant_mask_s16);
+    _vlog_ux = v_or(_vlog_ux, v_reinterpret_as_s16(v_setall_<_TpVec16F>(0.5f)));
```
Comment (WanliZhong) [inline modules/core/include/opencv2/core/hal/intrin_math.hpp:217]: The same as above

File: modules/core/include/opencv2/core/hal/intrin_neon.hpp [hunk 5]
```diff
@@ -435,6 +437,7 @@ OPENCV_HAL_IMPL_NEON_INIT(uint64x2, uint64, uint64, u64)
 OPENCV_HAL_IMPL_NEON_INIT(int64x2, int64, int64, s64)
 #if CV_SIMD128_FP16
 OPENCV_HAL_IMPL_NEON_INIT(float16x8, hfloat, __fp16,  f16);
+template <> inline v_float16x8 v_setall_(float v) { return v_setall_f16((hfloat)v); }
```
Comment (WanliZhong) [inline modules/core/include/opencv2/core/hal/intrin_neon.hpp:440]: Should be removed

File: modules/core/test/test_intrin_utils.hpp [hunk 6]
```diff
@@ -2084,6 +2090,99 @@ template<typename R> struct TheTest
...
+            }
+        }
+    }
+
+    TheTest &test_sincos_fp16() {
+#if CV_SIMD_FP16
+        hfloat flt16_min;
+        uint16_t flt16_min_hex = 0x0400;
+        std::memcpy(&flt16_min, &flt16_min_hex, sizeof(hfloat));
+        __test_sincos((hfloat) 1e-3, flt16_min);
```
Comment (WanliZhong) [inline modules/core/test/test_intrin_utils.hpp:2170]: release the threshold from `1e-3` to `1e-2`

**Review by WanliZhong — COMMENTED** [review]
I can't push commits to this PR. Could you help me to add these modifications?

**Comment by asmorkalov** [comment]
@WanliZhong Thanks for the review. All my changes are part of merge commit and do not contain description. I rpopose to not hide important changes, but just disable test in merge and add proper fix after merge.

**Comment by WanliZhong** [comment]
> @WanliZhong Thanks for the review. All my changes are part of merge commit and do not contain description. I rpopose to not hide important changes, but just disable test in merge and add proper fix after merge.

OK, then I will fix it after this brach merged.

**Pull request merged**